odoo_sh/advanced/submodules.rst

```diff
@@ -0,0 +1,104 @@
+
+Besides, you can choose the branch of the repository added as submodule
+and you have the control of the revision you want.
+It's up to you to decide wether you want to pin the submodule to a specific revision and when you want to update
+to a newer revision.
+
+In Odoo.sh, the submodules gives you the possibility to use and depends on modules available in other repositories.
+The platform will detect automatically that you added modules through submodules within your branches,
+and add them within your addons path so you can install them in your databases.
+
+If you add private repositories as submodules in your branches,
+you need to configure a deploy key in your Odoo.sh project settings and in your repository settings.
+Otherwise Odoo.sh won't be allowed to download them.
+The procedure is detailed in the chapter :ref:`Settings > Submodules <odoosh-gettingstarted-settings-submodules>`.
+
+Adding a submodule
+==================
+
+With Odoo.sh (simple)
+---------------------
+
+On Odoo.sh, in the branches view of your project, choose the branch in which you want to add a submodule.
+
+In the upper right corner, click on the *Submodule* button, and then on *Run*.
+
+.. image:: ./media/advanced-submodules-button.png
+   :align: center
+
+A dialog with a form is shown. Fill the inputs as follow:
+
+* Repository URL: The SSH URL of the repository.
+* Branch: The branch you want to use.
+* Path: The folder in which you want to add this submodule in your branch.
+
+.. image:: ./media/advanced-submodules-dialog.png
+   :align: center
+
+On Github, you can get the repository URL with the *Clone or download* button of the repository. Make sure to *use SSH*.
+
+.. image:: ./media/advanced-submodules-github-sshurl.png
+  :align: center
+
+At the moment, this is not possible to use this method to add private repositories.
+You can nevertheless do so :ref:`with Git <odoosh-advanced-submodules-withgit>`.
+
+.. _odoosh-advanced-submodules-withgit:
+
+With Git (advanced)
+---------------------
+
+In a terminal, in the folder where is cloned your Git repository,
+checkout the branch in which you want to add a submodule:
+
+.. code-block:: bash
+
+  $ git checkout <branch>
+
+Then, add the submodule using the below command:
+
+.. code-block:: bash
+
+  $ git submodule add -b <branch> <git@yourprovider.com:<username/repository.git> <path>
+
+Replace
+
+* *<git@yourprovider.com:<username/repository.git>* by the SSH URL of the repository you want to add as submodule,
+* *<branch>* by the branch you want to use in the above repository,
+* *<path>* by the folder in which you want to add this submodule.
+
+Commit and push your changes:
+
+.. code-block:: bash
+
+  $ git commit -a && git push -u <remote> <branch>
+
+Replace
+
+* <remote> by the repository on which you want to push your changes. For a standard Git setup, this is *origin*.
+* <branch> by the branch on which you want to push your changes.
+  Most-likely the branch you checkouted in the above steps.
```

Comment:
"checkouted"

I tried to ignore the above use of the verb "to checkout" but this one hurts my prefrontal cortex when I read it :p 

I know the command is `git checkout`, but the verb is "to check out", and thus checked out.

How about:

Most likely the branch you used `git checkout` on in the first step.